### DISCUSSION TITLE
Terraform lock files not updated if using Terragrunt

### DISCUSSION THREAD

**Discussion by tpansino** [renovatebot/renovate#13356] [Request Help]
I understand from reading the docs and various feature PRs that Terraform lock files are supported for lock file maintenance. Is this correct?

If so, I have the following repo:
```
myrepo/
├── dev
│   ├── .terraform.lock.hcl
│   └── terragrunt.hcl
├── prod
│   ├── .terraform.lock.hcl
│   └── terragrunt.hcl
└── stage
    ├── .terraform.lock.hcl
    └── terragrunt.hcl
```

And the following config:

```
{
  "extends": ["config:base"],
  "lockFileMaintenance": {
    "enabled": true
  },
  "packageRules": [
    {
      "matchUpdateTypes": ".*",
      "matchPackagePatterns": ".*",
      "groupName": "all dependencies",
      "groupSlug": "all"
    }
  ],
  "schedule": ["before 3am on Monday"],
  "separateMajorMinor": true,
  "timezone": "America/Los_Angeles"
}
```

The Terraform lock files aren't being updated, and I suspect it's because I'm using Terragrunt here.

So for the unfamiliar - Terragrunt essentially runs as a transpiler of Terraform. It allows for parameterization of parts of Terraform that Terraform otherwise requires be hard-coded. When you run Terragrunt, it generates a `.terragrunt-cache` directory in the current folder, containing the transpiled files, and then just calls Terraform within that directory. Terraform lock files are generated and checked into version control, but the cache files are not.

From reading the docs/code, it looks like:

- Terraform knows how to update a lock file if that lock file is in the same dir as the `*.tf` files containing the provider definitions
- Terragrunt doesn't know anything about lock files (because it doesn't have any of its own)

Which I think leaves a gap for updating Terraform lock files that were generated _via Terragrunt_ invocation.

What's my recourse here? Do I need a new manager that understands the cache dir?

**Comment by viceice** [comment]
Terraform and terragrunt are two different managers in renovate and only the terraform manager supports lockfile updates.

So you should crate a feature request for support lockfile updates on terragrunt.

I think we can reuse most of the logic from terraform.
